### DISCUSSION TITLE
Need some tips on how to prune sanoid snaps on source AND syncoid copied snaps on target

### DISCUSSION THREAD

**Discussion by ovizii** [jimsalterjrs/sanoid#887] [Q&A] [answered]
I'm sorry, but I am not sure I managed to adequately describe my question. Let me try and explain:

Say I have been running sanoid for a long time and have it configured as desired. It keeps a certain amount of hourly, daily, monthly and yearly snaps on the internal drives. 

I am now trying to figure out, how to do a periodic / manual run of syncoid to sync certain datasets to an external hard drive. 

I have managed this so far but I now realize that my external backup target is now a 1:1 copy of the source. This is not what I need. I would guesstimate that I would be pretty happy having 3 latest monthly snaps on my external backup target. (Open for other suggestions) 

I am now wondering how one best does this. I don't think running sanoid with autoprune to clean up the backup target makes sense as my current syncoid parameters woudl then copy all deleted snaps again. 

Currently this is my syncoid command:

`syncoid --source-bwlimit=90M --recursive --sendoption=w source/dataset target/dataset`

**Comment by jimsalterjrs** [comment, marked as answer]
> my current syncoid parameters woudl then copy all deleted snaps again.

That's not how replication works. Replication only adds snapshots to the 
*top* of the tree, not to the base.

Say you've got snapshots 0-8 on the source, which your replicate to the 
target. Now, you delete snapshots 0-7 on the target, and take a new 
snapshot 9 on the source.

After you next replicate to the target, the target will have snapshots 8 
and 9. It will not have 0-7, because replication can't go backwards--if you 
*wanted* to restore the missing snapshots 0-7, the only way to do it would 
be to destroy the entire backup and begin again with a full replication.

**Reply by ovizii** [reply]
Awesome, thanks for explaining. I will give it a try, worst case I'll just start fresh with my replication. 

One last question, I should have sanoid handle pruning for the external replciation target, right? There's no parameter for syncoid to do that after a replication? The reason I am asking is because I am unsure if sanoid will have any problems if I define pruning details in /etc/sanoid/sanoid.conf for a device which only shows up once a month when I run my replciation job and plug the external target in.

**Comment by jimsalterjrs** [comment]
You can keep a second configuration file for your external, and run sanoid 
manually with the -c argument (IIRC; this is not a feature I personally 
use) to point to that configuration. Something along the lines of:

sanoid -c /etc/sanoid/external.conf

**Reply by ovizii** [reply]
Does not seem to work. 
`sanoid -c /etc/sanoid/quad.conf`

result:
```
Option c is ambiguous (cache-dir, configdir, cron)
    Options:

      --configdir=DIR       Specify a directory to find config file sanoid.conf
      --cache-dir=DIR       Specify a directory to store the zfs snapshot cache

```
I do not see an option to specify a config file, just a directory to look for sanoid.conf. 
What am I missing?
Seeing that I also have the "default" `/etc/sanoid/sanoid.conf` do I really need an extra directory to copy and rename my special case quad.conf to?

**Reply by phreaker0** [reply]
Yes you need an extra dirctory.
